pkg/services/store/http.go

```diff
@@ -25,35 +27,49 @@ func ProvideHTTPService(store StorageService) HTTPStorageService {
 }
 
 func (s *httpStorage) Upload(c *models.ReqContext) response.Response {
-	action := "Upload"
-	scope, path := getPathAndScope(c)
+	// 32 MB is the default used by FormFile()
+	if err := c.Req.ParseMultipartForm(32 << 20); err != nil {
+		return response.Error(400, "error in parsing form", err)
+	}
+	res, err := s.store.Upload(c.Req.Context(), c.SignedInUser, c.Req.MultipartForm)
 
-	return response.JSON(200, map[string]string{
-		"action": action,
-		"scope":  scope,
-		"path":   path,
+	if err != nil {
+		return response.Error(500, "Internal Server Error", err)
+	}
+
+	return response.JSON(res.statusCode, map[string]string{
+		"message": res.message,
+		"path":    res.path,
+		"file":    res.fileName,
 	})
 }
 
 func (s *httpStorage) Read(c *models.ReqContext) response.Response {
-	action := "Read"
-	scope, path := getPathAndScope(c)
-
-	return response.JSON(200, map[string]string{
-		"action": action,
-		"scope":  scope,
-		"path":   path,
-	})
+	// full path is api/storage/read/upload/example.jpg, but we only want the part after read
+	indexOfPath := strings.Index(c.Req.RequestURI, "read") + 4
```

Comment:
I think we can just read the "*" parameter like we do here:
https://github.com/grafana/grafana/blob/b4eb8c2b3aedaefcaa2009572b8d66297fe933dd/pkg/services/store/http.go#L78